src/stream.rs

```diff
@@ -20,55 +23,92 @@ pub enum ProxyStream<R> {
     Secured(TlsStream<R>),
 }
 
-macro_rules! match_fn {
-    ($self:expr, $fn:ident $(,$buf:expr)*) => {
-        match *$self {
-            ProxyStream::Regular(ref mut s) => s.$fn($($buf)*),
+macro_rules! match_fn_pinned {
+    ($self:expr, $fn:ident, $ctx:expr, $buf:expr) => {
+        match $self.get_mut() {
+            ProxyStream::Regular(s) => Pin::new(s).$fn($ctx, $buf),
             #[cfg(any(feature = "tls", feature = "rustls"))]
-            ProxyStream::Secured(ref mut s) => s.$fn($($buf)*),
+            ProxyStream::Secured(s) => Pin::new(s).$fn($ctx, $buf),
         }
-    }
-}
-
-impl<R: Read + Write> Read for ProxyStream<R> {
```

Comment:
Please note this Read/Write trait implementation removal. TlsStream not implements them anymore